tests/base.py

```diff
@@ -164,23 +177,22 @@ def setUp(self):
             TestProgress(2, TestStatus.completed, "Test 2 completed")
         ]
         g.db.add_all(test_result_progress)
+        g.db.commit()
```

Comment:
No need to add commit after every addition. You could add this at the end and all additions will be committed.

Reply:
I guess committing first is right. If we don't commit it first, then we will get foreign key errors. as we are using "id" while creating a new row of other table.

Reply:
@satyammittal Okay, if you've tested it :)